### PR TITLE
Change OSError and its subclasses attributes

### PR DESCRIPTION
**Pull request by ChJR**
This PR provides specified attribute "characters_written" to BlockingIOError. Other unnecessary attributes are ignored in BlockingIOError such as winerror, filename, filename2.

### CODE REVIEW

**Comment by DimitrisJim** [comment]
Related: #2974. Haven't been paying much attention to it but you might want to see the updates there since they should affect how this PR should try and implement this.

**Review by youknowone — DISMISSED** [review]

**Comment by youknowone** [comment]
@deantvv could you review this PR?

File: vm/src/exceptions.rs [hunk 1]
```diff
@@ -1227,29 +1210,82 @@ pub(super) mod types {
...
-        PyBaseException::slot_new(cls, args, vm)
+        let exc_self = PyBaseException::new(args.args.to_vec(), vm)
+            .into_ref_with_type(vm, cls)
+            .unwrap();
+        os_error_parse_args(exc_self.clone(), &args, vm)?;
+        exc_self.into_pyresult(vm)
+    }
+
+    fn os_error_parse_args(
+        exc_self: PyBaseExceptionRef,
```
Comment (deantvv) [inline vm/src/exceptions.rs:1240]:
```suggestion
        exc_self: &PyBaseExceptionRef,
```

Comment (deantvv) [reply vm/src/exceptions.rs:1240]: We could use the reference to remove some `clone`.

**Review by deantvv — COMMENTED** [review]
This PR looks great and makes OSError behaves more like CPython version.

**Comment by ChJR** [comment]
Too old to fix the codes. I close this PR and will check other issues.

**Pull request closed without merging**